### PR TITLE
Fix to include TF_ROCM_VERSION>=40xxx branches in HipSolver and HipSparse

### PR DESCRIPTION
**Pull request by stevenireeves**

### CODE REVIEW

File: tensorflow/compiler/xla/service/gpu/cusolver_context.cc [hunk 1]
```diff
@@ -19,6 +19,9 @@ limitations under the License.
 #include <complex>
 #include <utility>
 
+#if TENSORFLOW_USE_ROCM
```
Comment (rsanthanam-amd) [inline tensorflow/compiler/xla/service/gpu/cusolver_context.cc]: I don't think this is necessary because it is in cusolver_context.h.

Comment (stevenireeves) [reply tensorflow/compiler/xla/service/gpu/cusolver_context.cc]: Removed.

**Comment by deven-amd** [comment]
@stevenireeves , can I request you to re-organize the changes into the following commits

1. revert commit - https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/pull/1607/commits/169221aa95b10573529e6f943c7369c6c0d56f6c
2. revert commit - https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/pull/1612/commits/228c12145d86ef46ead1cd9a6df557ddd18e8b99
3. commit to include the `rocm/rocm_config.h` header in all files that have reference to TF_ROCM_VERSION.  
4. commit for changes related to `hipsolver` / `rocsolver`  to fix build failures
5. commit for changes related to `hipsparse` to fix build failures 

thanks

**Comment by deven-amd** [comment]
I am running into `hipsparse` related build failures when I try to backport my changes to the `r2.7-rocm-enhanced` and `r2.8-rocm-enhanced` branches. Those branches have the `hipsparse` related updates, but not the `hipsolver` related ones, and hence the request to break them apart...makes it easier to cherry-pick just the necessary bits.

**Comment by stevenireeves** [comment]
@deven-amd these are ready. Let me know if you need help with r2.7/2.8

**Comment by deven-amd** [comment]
thank you @stevenireeves .

let me cherry-pick these commits to the `r.27-rocm-enhanced` and `r2.8-rocm-enhanced` branches, and see if they make them pass

**Comment by stevenireeves** [comment]
Jenkins : retest Ubuntu-GPU-multi please

**Pull request merged**